src/schema/loader.ts

```diff
@@ -0,0 +1,25 @@
+// Import Node.js Dependencies
+import { readdirSync } from "node:fs";
+import path from "node:path";
+
+// Import Internal Dependencies
+import { readJSONSync } from "../utils/index.js";
+
+// CONSTANTS
+const kDefsDirectory = new URL("./defs", import.meta.url);
+
+function definitionReducer($defs: Record<string, any>, fileName: string) {
```

Comment:
We could probably name this function with a name reflecting its intent rather than its technical implementation, what do you think?